prow/pjutil/pjutil.go

```diff
@@ -175,6 +176,15 @@ func EnvForSpec(spec kube.ProwJobSpec) map[string]string {
 		"JOB_NAME": spec.Job,
 	}
 
+	rawSpecData, err := json.Marshal(spec)
+	if err != nil {
+		// there should be no point at which these objects
+		// cannot be serialized, or we couldn't store them
+		// in the apiserver in the first place
+		panic(err)
```

Comment:
Even though your comment is correct, still propagate up the error. Panicking will cause dropped webhooks.

Reply:
Yeah, was going to ask how you wanted to handle this